hat/app/org/hatdex/hat/api/controllers/RichData.scala

```diff
@@ -679,12 +679,71 @@ class RichData @Inject() (
           case Left(PublicKeyRequestFailure.InvalidPublicKeyFailure(failureDescription)) => {
             Left(ContractVerificationFailure.ServiceRespondedWithFailure(s"The Adjudicator Service responded with an error: ${failureDescription}"))
           }
-          case Right(PublicKeyReceived(publicKey)) => verifyJwyClaim(contractRequestBodyRefined, publicKey)
+          case Right(PublicKeyReceived(publicKey)) => verifyJwtClaim(contractRequestBodyRefined, publicKey)
         }
       }
     }
   }
 
+
+
+  /* *** Contract Data *** */
+
+  def getContractData(
+    namespace: String,
+    endpoint: String,
+    orderBy: Option[String],
+    ordering: Option[String],
+    skip: Option[Int],
+    take: Option[Int]): Action[ContractRequestBody] =
+    UserAwareAction.async(parsers.json[ContractRequestBody]) { implicit request =>
+      val contractRequestBody = request.body
+      val requestIsAllowed = assessRequest(contractRequestBody, namespace)
+
+      requestIsAllowed.flatMap { testResult =>
+        testResult match {
+          case Right(RequestVerified(ns)) => makeData(namespace, endpoint, orderBy, ordering, skip, take)
+          case _ => Future.successful(NotFound)
```

Comment:
Would it be worth logging the error ?
Not returning VIA the API is a good idea for security reasons but the error that occurred is getting swallowed here